### PR TITLE
Relax constraint `num_bits(storage_type) < num_bits(expressed_type)`

### PR DESCRIPTION
**Pull request by sdasgup3**
fixes https://github.com/openxla/stablehlo/issues/2205

Requesting the following for review:  
@loganchien
@paulinesho
@doyeonkim0

### CODE REVIEW

File: docs/spec.md [hunk 1]
```diff
@@ -158,18 +158,17 @@ in the grammar, but have default values of `min_value(storage_type)` and
 `max_value(storage_type)` respectively. Quantized element types have the
 following constraints:
 
-* (C1) `num_bits(storage_type) < num_bits(expressed_type)`.
```
Comment (loganchien) [inline docs/spec.md:161]: Need to update `C_i` at Line 143-149?

Comment (sdasgup3) [reply docs/spec.md:161]: Thanks for pointing that. fixed.

**Review by loganchien — APPROVED** [review]

**Comment by doyeonkim0** [comment]
Out of curiosity, do we have use case for `num_bits(storage_type) > num_bits(expressed_type)`?

**Comment by GleasonK** [comment]
> Out of curiosity, do we have use case for `num_bits(storage_type) > num_bits(expressed_type)`?

+1

**Comment by sdasgup3** [comment]
@doyeonkim0 
Per my offline chat with Andy, we might also have `i32` storage type and `bf16` expressed type for the same use case for matching the output type of convolution.

**Comment by doyeonkim0** [comment]
That makes sense. Thanks for the answer!

**Review by doyeonkim0 — APPROVED** [review]

**Review by GleasonK — APPROVED** [review]
This feels in line with the proposal in:
https://github.com/openxla/stablehlo/blob/main/rfcs/20230622-quantized-reduction.md

Even in that proposal you use examples where the bitwidths violate this constraint as written (i.e. `!qunat.uniform<i32:f32, ...>`). Given that that proposal was approved, I think this is a reasonable spec correction.

**Pull request merged**